tee-worker/ts-tests/utils.ts

```diff
@@ -141,161 +144,47 @@ export async function sendTxUntilInBlock(api: ApiPromise, tx: SubmittableExtrins
+    return new Promise<Event[]>(async (resolve, reject) => {
         let startBlock = 0;
-        const slotDuration = await context.substrate.call.auraApi.slotDuration();
-        const timeout = 3 * 60 * 1000; // 3 min
-        let maximumWaitingBlock = timeout / parseInt(slotDuration.toString());
-        console.log('maximumWaitingBlock', maximumWaitingBlock);
-        const unsubscribe = await context.substrate.rpc.chain.subscribeNewHeads(async (header) => {
+        const unsubscribe = await api.rpc.chain.subscribeNewHeads(async (header) => {
             const currentBlockNumber = header.number.toNumber();
             if (startBlock == 0) startBlock = currentBlockNumber;
-            if (currentBlockNumber > startBlock + maximumWaitingBlock) {
+            if (currentBlockNumber > startBlock + listenTimeoutInBlock) {
                 reject('timeout');
                 return;
             }
-            console.log(`Chain is at block: #${header.number}`);
-            const signedBlock = await context.substrate.rpc.chain.getBlock(header.hash);
-            const allEvents = (await context.substrate.query.system.events.at(header.hash)) as Vec<EventRecord>;
-            signedBlock.block.extrinsics.forEach((ex, index) => {
-                if (!(ex.method.section === 'identityManagement')) {
-                    return;
-                }
-                const list = allEvents
-                    .filter(({ phase, event }) => {
-                        return (
-                            phase.isApplyExtrinsic &&
-                            event.section == 'identityManagement' &&
-                            (event.method == 'IdentityCreated' || event.method == 'ChallengeCodeGenerated')
-                        );
-                    })
-                    .map(({ event }) => {
-                        const data = event.data as AESOutput[];
-                        return data;
-                    });
-                if (list.length) {
-                    const eventData: HexString[] = [];
-                    for (let i = 0; i < list.length; i++) {
-                        for (let j = 0; j < list[i].length; j++) {
-                            eventData.push(decryptWithAES(aesKey, list[i][j]));
-                        }
-                    }
-                    resolve({ eventData: [...new Set(eventData)] });
-                    unsubscribe();
-                    return;
-                }
+            console.log(`\n--------- block #${header.number}, hash ${header.hash} ---------\n`);
+            const apiAt = await api.at(header.hash);
+
+            const records: EventRecord[] = (await apiAt.query.system.events()) as any;
+            records.forEach((e, i) => {
+                const s = (e.toHuman() as any).event.section;
+                const m = (e.toHuman() as any).event.method;
+                const d = e.event.data;
+                console.log(`Event[${i}]: ${s}.${m} ${d}`);
```

Comment:
Why not change these code to 
```ts
const records = (await apiAt.query.system.events()) as Vec<EventRecord>;
records.forEach((e, i) => {
    const s = e.event.section;
    const m = e.event.method;
    const d = e.event.data;
    console.log(`Event[${i}]: ${s}.${m} ${d}`);
});
```

Reply:
Me too.

Reply:
Thanks, changed. It didn't work out in my first try, therefore I changed it - but now it seems working again, strange 😂